core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -332,6 +332,36 @@ class SparkSubmitSuite
     sys.props("SPARK_SUBMIT") should be ("true")
   }
 
+  test("handles spark.yarn.applicationType on yarn mode") {
+    val clArgs = Seq(
+      "--deploy-mode", "client",
+      "--master", "yarn",
+      "--executor-memory", "5g",
+      "--executor-cores", "5",
+      "--class", "org.SomeClass",
+      "--jars", "one.jar,two.jar,three.jar",
+      "--driver-memory", "4g",
+      "--queue", "thequeue",
+      "--files", "file1.txt,file2.txt",
+      "--archives", "archive1.txt,archive2.txt",
+      "--num-executors", "6",
+      "--name", "trill",
+      "--conf", "spark.yarn.applicationType=SPARK-SQL",
```

Comment:
should you also test the case when `spark.yarn.applicationType` is not set?

Reply:
I agree, what happens if empty string, any characters not allowed in it? Add tests and documentation to cover.

Reply:
you didn't add tests for other like one empty string?  
Also I believe the limit on that string is 20 characters, we should test and document that. Can you also test a space in the type?